### PR TITLE
Dynamic selection tests improvements

### PR DESCRIPTION
**Pull request by SergeyKopienko**
Now it's only draft PR.
If you think that this tests state is better and we are going to apply this PR, I am ready to compile m run and check all this tests.

### CODE REVIEW

**Review by timmiesmith — CHANGES_REQUESTED** [review]

File: test/parallel_api/dynamic_selection/test_internal_inline_scheduler.pass.cpp [hunk 1]
```diff
@@ -7,210 +7,235 @@
...
+int
+test_submit_and_wait_on_sync_empty()
+{
+    const int N = 0;
+    TestUtils::int_inline_backend_t s;
+    fake_selection_handle_t h;
+
+    std::atomic<int> ecount = 0;
+
+    for (int i = 1; i <= N; ++i)
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/parallel_api/dynamic_selection/test_internal_inline_scheduler.pass.cpp:184]: Good point.
It's from the initial commit "Added experimental feature for dynamic device selection" :
https://github.com/oneapi-src/oneDPL/commit/ea466d230fc4f74ccd298f752a2201b08a0900dc

[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/parallel_api/dynamic_selection/test_internal_inline_scheduler.pass.cpp:184]: Also may be interested the correctness of all another assignments like `const int N = ` in dynamic selection tests.
Because there are assignments to `0`, `1`, `100` and etc.

File: test/parallel_api/dynamic_selection/test_internal_inline_scheduler.pass.cpp [hunk 2]
```diff
@@ -7,210 +7,235 @@
...
+int
+test_submit_and_wait_on_scheduler_empty()
+{
+    const int N = 0;
+    TestUtils::int_inline_backend_t s;
+    fake_selection_handle_t h;
+
+    std::atomic<int> ecount = 0;
+
+    for (int i = 1; i <= N; ++i)
```
[the remark this answers is not in the document]
Comment (egfefey) [reply test/parallel_api/dynamic_selection/test_internal_inline_scheduler.pass.cpp]: This is a test for an empty submit so N being 0 is fine. Maybe it's better to remove the loop to avoid confusion.

File: test/parallel_api/dynamic_selection/test_internal_inline_scheduler.pass.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (egfefey) [reply test/parallel_api/dynamic_selection/test_internal_inline_scheduler.pass.cpp]: This is also a test for an empty submit but the wait is inside the loop so that has to be fixed. Might also be better to just remove the loop.

**Review by egfefey — APPROVED** [review]
@SergeyKopienko 
I have updated the device selector part so it should be Ok to merge.
I did not merge it because there are also some pstl tests. If the pstl tests are Ok, then you can merge the PR.

File: test/parallel_api/dynamic_selection/test_internal_inline_scheduler.pass.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (egfefey) [reply test/parallel_api/dynamic_selection/test_internal_inline_scheduler.pass.cpp]: Fixed

File: test/parallel_api/dynamic_selection/test_internal_inline_scheduler.pass.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (egfefey) [reply test/parallel_api/dynamic_selection/test_internal_inline_scheduler.pass.cpp]: Fixed

**Comment by SergeyKopienko** [comment]
@egfefey could you please create your own git branch based on `https://github.com/oneapi-src/oneDPL/tree/dev/skopienko/dynamic_selection_tests` or new branch from `main` and copy all required and copy all required from your point of view changes into this new branch? After that please create your own PR? And we will merge it...
Thanks.
